proposals/lip-Introduce_cross-chain_update_mechanism.md

```diff
@@ -0,0 +1,620 @@
+
+
+#### Cross-chain Updates Posted on Sidechains
+
+Cross-chain update transaction posted on sidechains are transactions with
+
+*   `moduleID = MODULE_ID_INTEROPERABILITY`,
+*   `commandID  = COMMAND_ID_SIDECHAIN_CCU`. 
+
+To execute cross-chain updates, the following is done:
+
+*   If `partnerChain.status == CHAIN_REGISTERED`, set `partnerChain.status = CHAIN_ACTIVE`.
+*   For every `CCM` in `inboxUpdate.crossChainMessages`:
+    *   Validate that `ownChainID == CCM.receivingChainID`.
+    *   Validate the format of `CCM` according to the function provided in [LIP "Introduce cross-chain messages"][CCM-LIP]. 
+*   Validate that the first CCM in `inboxUpdate.crossChainMessages` has `CCM.index == partnerChain.inbox.size`.
+*   Validate that all CCMs in `inboxUpdate.crossChainMessages` have increasing and sequential index property.
+*   If one of those validation points fails:
+    *   `terminateChain(partnerChainID)` as specified in [LIP "Introduce Interoperability module"][base-interoperability-LIP].
+    *   Exit the transaction processing, the CCU has no further effect.
+*   Continue the processing by executing all steps in the ["Common Processing" section](#common-processing).
+
+
+#### Common Processing
+
+For CCU transactions posted on the mainchain or on sidechains, once the specific execution steps described above have been completed, the following is done in the given order:
+
+*   For every `CCM` in `inboxUpdate.crossChainMessages` with  `CCM.receivingChainID == ownChainID`:
+    *   Assign `CCM.fee` to the relayer, here `relayerAddress` is the address corresponding to the sender public key of the CCU. This is done with the [token function][token-LIP] `beforeExecuteCCM(relayerAddress, CCM)`.
+*   For every `CCM` in `inboxUpdate.crossChainMessages` (respecting the order of the array):
+    *   Call <code>[appendToInboxTree][base-interoperability-LIP-appendToInboxTree](partnerChainID, SHA-256(serializedMessage))</code> where `serializedMessage` is the serialized CCM according to the schema given in [LIP "Introduce cross-chain messages"][CCM-LIP].
+    *   Process `CCM` as detailed in [LIP "Introduce cross-chain messages"][CCM-LIP-process].
+*  Update `partnerChain.validators` according to `validatorsUpdate`, see ["Update Validators" section](#update-validators).
+*  Set `partnerChain.lastCertifiedStateRoot` to `certificate.stateRoot`.
+*  Set `partnerChain.lastCertifiedTimestamp` to `certificate.timestamp`.
+*  Set `partnerChain.lastCertifiedHeight` to `certificate.height`.
+*  Set `partnerChain.partnerChainOutboxSize` to
+    *   `inboxUpdate.messageWitness.partnerChainOutboxSize` if `params` contains a non-empty `inboxUpdate.messageWitness`.
+    *   `partnerChain.inbox.size` otherwise.
+*   Set `partnerChain.partnerChainOutboxRoot` to
+    *   `calculateRootRightWitness(inbox.size, inbox.appendPath, inboxUpdate.messageWitness.siblingHashes)` if `params` contains a non-empty `inboxUpdate.messageWitness`.
```

Comment:
The function `calculateRootRightWitness` here is undefined. Maybe we could add a link to the Merkle tree LIP.

Reply:
Also the function name is `calculateRootFromRightWitness`

Reply:
changed name.
added link.